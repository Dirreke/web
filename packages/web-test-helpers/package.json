{
  "name": "web-test-helpers",
  "private": true,
  "peerDependencies": {
      "@vue/composition-api": "1.7.1",
      "vue-gettext": "^2.1.5",
      "vue-router": "3.6.5",
      "vuex": "^3.6.2"
  }
}
