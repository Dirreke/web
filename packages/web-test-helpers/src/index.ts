export * from './mocks/defaultComponentMocks'
export * from './mocks/defaultStubs'
export * from './mocks/axios'

export * from './defaultPlugins'
export * from './helpers'
