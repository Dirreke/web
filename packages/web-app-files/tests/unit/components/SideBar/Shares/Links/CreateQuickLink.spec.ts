import CreateQuickLink from 'web-app-files/src/components/SideBar/Shares/Links/CreateQuickLink.vue'
import { defaultPlugins, mount } from 'web-test-helpers'

const selectors = { createBtn: '.link-name-container button' }

describe('CreateQuickLink', () => {
  it('renders the quick link button', () => {
    const { wrapper } = createWrapper()
    expect(wrapper.find(selectors.createBtn).exists()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('emits the "createPublicLink"-action', async () => {
    const { wrapper } = createWrapper()
    const emitSpy = jest.spyOn(wrapper.vm, '$emit')
    await wrapper.find(selectors.createBtn).trigger('click')
    expect(emitSpy).toHaveBeenCalledWith('createPublicLink', expect.anything())
  })
})

function createWrapper({ props = {} } = {}) {
  return {
    wrapper: mount(CreateQuickLink, {
      props: { expirationDate: {}, ...props },
      global: { plugins: [...defaultPlugins()] }
    })
  }
}
