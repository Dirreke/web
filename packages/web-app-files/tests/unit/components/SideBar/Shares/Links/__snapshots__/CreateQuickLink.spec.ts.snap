// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateQuickLink renders the quick link button 1`] = `
<div class="oc-mb-s oc-width-1-1 oc-mb-l">
  <h4 class="oc-text-truncate oc-text-normal oc-files-file-link-name oc-my-rm">Quick link</h4>
  <div class="oc-flex oc-flex-middle oc-flex-between oc-width-1-1 oc-p-s link-name-container">
    <div class="oc-flex oc-flex-middle oc-text-truncate">
      <span class="oc-icon oc-icon-m oc-icon-passive">
        <!---->
      </span>
      <p class="oc-files-file-link-url oc-ml-s oc-text-truncate oc-my-rm">No link</p>
    </div>
    <button aria-label="Create quick link" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw" type="button">Create link</button>
  </div>
</div>
`;
